pkg/compactor/compactor_test.go

```diff
@@ -1364,6 +1364,129 @@ func TestMultitenantCompactor_ShouldSkipCompactionForJobsNoMoreOwnedAfterPlannin
 	))
 }
 
+func TestMultitenantCompactor_ShouldSkipCompactionForJobsWithFirstLevelCompactionBlocksAndWaitPeriodNotElapsed(t *testing.T) {
+	t.Parallel()
+
+	const waitPeriod = 10 * time.Minute
+
+	// Mock two tenants, each with the same exact overlapping blocks. However,
+	// the 2nd tenant has uploaded the blocks less than "wait period" ago.
+	bucketClient := &bucket.ClientMock{}
```

Comment:
[nit] Could we use a real bucket instead? (inmem or filesystem) These "mocks" create brittle tests that are tightly coupled with the implementation (eg. we need to provide response for deletion-mark.json or no-compact-mark.json files, even if they are not important for this test at all).

Reply:
Good point. Done in https://github.com/grafana/mimir/pull/4401/commits/6e8f723a12df1fdf6054d1e4faf10372898570c6.